### PR TITLE
Picard HsMetrics: Support any custom X coverage metrics

### PR DESCRIPTION
**Pull request by vladsavelyev**

### CODE REVIEW

File: multiqc/modules/picard/HsMetrics.py [hunk 1]
```diff
@@ -346,9 +335,26 @@ def _generate_table_header_config(table_cols, hidden_table_cols):
             for s, r in title_cleanup:
                 h_title = h_title.replace(s, r)
 
+            # Extract the coverage from the column name
+            m = re.match(r".+_(\d+)X", h)
+            if m:
+                try:
+                    cov = int(m.group(1))
+                except ValueError:
```
Comment (ewels) [inline multiqc/modules/picard/HsMetrics.py:343]: Given that the regex pattern is `\d+` is this not guaranteed to always work?

**Review by ewels — COMMENTED** [review]
Untested, but LGTM 👍🏻 

Note that the other Picard modules do some very similar things. May be worth bringing some functionality out into a Picard `utils.py` and sharing logic 🤔  Or at least mirroring how this works to the other ones.

Needs a docs update to the picard docs too.

File: multiqc/modules/picard/HsMetrics.py (hunk 1 above)
Comment (vladsavelyev) [reply multiqc/modules/picard/HsMetrics.py]: Good point! Simplified the code

**Comment by ewels** [comment]
`PF_BASES` is also a new field - this should be hardcoded into the field descriptions.

**Pull request merged**